src/directLine.test.ts

```diff
@@ -61,3 +65,223 @@ describe("#commonHeaders", () => {
         });
     })
 });
+
+describe("MockSuite", () => {
+
+    const lazyConcat = <T>(items: Iterable<Observable<T>>): Observable<T> =>
+        new Observable<T>(subscriber => {
+            const iterator = items[Symbol.iterator]();
+            let inner: Subscription | undefined;
+
+            const pump = () => {
+                try {
+                    const result = iterator.next();
+                    if (result.done === true) {
+                        subscriber.complete();
+                    }
+                    else {
+                        inner = result.value.subscribe(
+                            value => subscriber.next(value),
+                            error => subscriber.error(error),
+                            pump);
+                    }
+                }
+                catch (error) {
+                    subscriber.error(error);
+                }
+            };
+
+            pump();
+
+            return () => {
+                if (inner !== undefined) {
```

Comment:
I need to read up some articles about their difference. But I see people usually do it this way, must be for some reasons.

```suggestion
                if (typeof inner !== 'undefined') {
```

Reply:
You can change the value of undefined in older browsers. :)